pkg/pillar/cmd/monitor/monitor.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package monitor
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/agentbase"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	agentName            = "monitor"
+	errorTime            = 3 * time.Minute
+	warningTime          = 40 * time.Second
+	stillRunningInterval = 25 * time.Second
+)
+
+var logger *logrus.Logger
+var log *base.LogObject
+
+type monitorContext struct {
+	agentbase.AgentBase
+
+	subscriptions       map[string]pubsub.Subscription
+	pubDevicePortConfig pubsub.Publication
+	clientConnected     chan bool
+	serverNameAndPort   string
+	// cache last known data structures to avoid sending duplicate messages
+	lastNodeStatus *NodeStatus
+
+	IPCServer *IPCServer
+}
+
+func (ctx *monitorContext) readServerFile() error {
+	if _, err := os.Stat(types.ServerFileName); os.IsNotExist(err) {
+		ctx.serverNameAndPort = ""
+		return nil
+	}
```

Comment:
TOCTOU

better to just check if `err` from `os.Readfile` is an ENOENT

Reply:
os.IsNotExist is the golang way to check this AFAIK.

Reply:
https://pkg.go.dev/os#IsNotExist

> This function predates [errors.Is](https://pkg.go.dev/errors#Is). It only supports errors returned by the os package. New code should use errors.Is(err, fs.ErrNotExist).

good that we talked about it :-)